pages/open_cas_linux/guide_tool_details.md

```diff
@@ -758,6 +759,12 @@ Threshold is expressed in KiB.
 **[-j, -\-core-id \<ID\>]**: Unique identifier for core \<0 to 4095\>. If not
 specified, core parameter is set to all cores in a given cache.
 
+**[-\-promotion-count \<NUMBER\>]**: Specify how many requests an IO stream
+from a single thread is expected to be producing before this thread is
+rescheduled to another CPU. If the count is reached, the stream is promoted
+to global structures for future reference. This prevents rare streams to be
+promoted and therefore make less attempts to access shared structures.
+
```

Comment:
I suggest to modify this description so that it starts with a description of what the parameter controls, instead of how it is meant to be used. Also I am missing information on the nature and purpose of stream promotion. I suggest the following:

Specify number of IO operations required for a sequential stream tracking information to be moved from per-CPU storage to global storage. This process of moving tracking information to global context is called stream promotion. Once a stream is promoted, CAS will continue recognize its continuation even after application gets migrated to another CPU. If a stream is migrated to another CPU before it gets promoted, it will not be recognized as a continuation an it will be treated as a new stream instead. The lower the promotion threshold, the more synchronization between processors occurs, potentially affecting IO performance. On other hand the higher the promotion threshold, the more likely it is for CAS to lose track of sequential stream due to its origin application being moved to another CPU. Recommendation is to set this parameter to value low enough so that an I/O stream is expected to send this many I/O requests before being rescheduled to another CPU.